dothething/dothething/ClipsView.swift

```diff
@@ -35,71 +34,101 @@ struct ClipsView: View {
-                .padding(.leading)
-                .frame(maxWidth: .infinity, alignment: .leading)
+        GeometryReader { geometry in
+            VStack {
+                ZStack {
+                    HStack {
+                        Button(action: {
+                            print("Back button tapped")
+                            clipsViewModel.backButtonPressed()
+                            toggle.toggle()
+                        }) {
+                            Image(systemName: "chevron.left")
+                                .font(.system(size: 24))
+                                .foregroundColor(.accentColor)
+                        }
+                        .padding(.leading)
+                        Spacer()
+                    }
+                    
+                    Text("domino")
+                        .font(.custom("Montserrat-Medium", size: 27))
+                        .foregroundColor(Color.accentColor)
+                        .tracking(8)
+                        .multilineTextAlignment(.center)
+                }
                 
-                // upload button
-                Button(action: {
-                    clipsViewModel.uploadButtonPressed()
-                }) {
-                    Text("Upload")
+                Text("rally")
+                    .font(.custom("Montserrat-Medium", size: 17))
+                    .padding(.bottom)
+                    .tracking(4)
+
+                if !clipsViewModel.errorText.isEmpty && clipsViewModel.clips.isEmpty {
+                    Text(clipsViewModel.errorText)
+                        .padding(.top)
+                        .foregroundColor(Color(UIColor.systemGray))
+                } else if clipsViewModel.isLoading {
+                    ProgressView()
+                        .progressViewStyle(.circular)
+                        .padding(.top)
                 }
-                .opacity(clipsViewModel.uploadDisabled ? 0.5 : 1)
-
-                // share button
-                Button(action: {
-                    clipsViewModel.shareButtonPressed()
-                }) {
-                    Text("Share")
+                
+                ScrollView {
+                    LazyVGrid(columns: [GridItem](repeating: GridItem(.flexible(), spacing: 4), count: 3), spacing: 4) {
+                        ForEach(clipsViewModel.clips, id: \.self) { clip in
+                            ClipView(clip: clip)
+                        }
+                        .frame(height: (192 / 108) * geometry.size.width / 3)
+                    }
+                    .padding(.leading, 4)
+                    .padding(.trailing, 4)
                 }
-                .padding(.trailing)
-                .frame(maxWidth: .infinity, alignment: .trailing)
-                .opacity(clipsViewModel.shareDisabled ? 0.5 : 1)
```

Comment:
The deleted code here appears to be responsible for displaying an upload button and a share button in the UI, which would allow the user to perform those actions.